internal/repo/client/metaclient.go

```diff
@@ -65,6 +66,27 @@ func (d metaclient) FetchMetaIndex(target *types.MetaIndex) error {
 	}
 }
 
+func (d metaclient) FetchMetaIndexForRepo(repo string, target *types.MetaIndex) error {
```

Comment:
The logic here seems sound, but I think it's confusing to have a function for a specific repo on the metaclient. Since this feature is quite specific to the list package, I think that's where it should be kept. Can you please move this function to `pkg/list`, such that it can be called like this (you can choose a different name if you want):

```go
var index types.PackageRepoIndex
d.clientset.ForRepoWithName(repo).FetchPackageRepoIndex(&index)
metaIndex := indexToMetaIndex(index)
```

Reply:
makes sense, I have modified as suggested.